spotify/spotify-web-api-ts-sdk
# Update return types for getPlaylist, getPlaylistItems

### Summary
Updated return types for `getPlaylist`, `getPlaylistItems` to reflect status of `additional_types` parameter.
Both methods will now return types with `PlaylistedTrack<Track>` if `additional_types` is omitted.
If `additional_types` is included, the resulting type will included `Episode` as `PlaylistTrack<TrackItem>` (where `type TrackItem = Track | Episode`).

### Changes
- Added `TrackItem` alias for all playable item types in playlists and player state
- Added `QueryAdditionalTypes` alias for `additional_types` parameters (definition from OpenAPI spec states only `'episode'` is a valid value at this moment)
- Updated `PlaylistedTrack` and `Playlist` types to be generic to support narrowed item types
- Updated `getPlaylist`, `getPlaylistItem` with generics to conditionally use `Track` or `TrackItem` in return types
- Added tests for `getPlaylist`, `getPlaylistItem` where `additional_types` is passed
- Fixed various typos

### Testing
- Check that the newly added tests pass
- Check that the return type for `getPlaylist` is now `Promise<Playlist<Track>>` by default and `Promise<Playlist<TrackItem>>` when `additional_types` is passed
- Check that the return type for `getPlaylistItems` is now `Promise<Page<PlaylistedTrack<Track>>>` by default and `Promise<Page<PlaylistedTrack<TrackItem>>>` when `additional_types` is passed
